app/src/state/editor.ts

```diff
@@ -146,6 +161,58 @@ export const saveDocument = createAsyncThunk<Document, void, { state: RootState
     return document;
   }
 );
+
+export const exportDocument = createAsyncThunk<Document, void, { state: RootState }>(
+  'editor/exportDocument',
+  async (_, { dispatch, getState }) => {
+    const document = getState().editor.present?.document;
+    if (document === undefined) {
+      throw Error('cant export. document is undefined');
+    }
+    console.log('starting export', document.content);
+    const combined_content = [];
+    let cur_start = 0;
+    let cur_end = 0;
+    let cur_source: number | null = null;
+    document.content.forEach((paragraph) => {
```

Comment:
this might be usefull to have in a seperate function, since the player would need to do the same thing to avoid crackling noises